sql/src/main/java/org/opensearch/sql/sql/SQLService.java

```diff
@@ -65,10 +65,15 @@ private AbstractPlan plan(
       SQLQueryRequest request,
       Optional<ResponseListener<QueryResponse>> queryListener,
       Optional<ResponseListener<ExplainResponse>> explainListener) {
+    boolean isExplainRequest = request.isExplainRequest();
     if (request.getCursor().isPresent()) {
       // Handle v2 cursor here -- legacy cursor was handled earlier.
-      return queryExecutionFactory.createContinuePaginatedPlan(request.getCursor().get(),
-          request.isExplainRequest(), queryListener.orElse(null), explainListener.orElse(null));
+      if (isExplainRequest) {
+        throw new UnsupportedOperationException("Explain of a paged query continuation " +
+            "is not supported. Use `explain` for the initial query request.");
```

Comment:
Why it is here?

Reply:
If we don't check for explain of cursor request, the user gets and error message about invalid cursor. This exception was already being thrown from a different class but that class was removed and the check needed a new home.

Explain in legacy does not work as well so there isn't loss in functionality.

Reply:
I think this class isn't a right place of generating response (this exception returned to a user). Probably, you can omit this and let cursor explain query to be processed as explain for a paged request. Take into account that `create` gets `isExplainRequest` argument and creates a proper explain plan for that.